src/hooks/useSubaccount.tsx

```diff
@@ -980,7 +980,7 @@ const useSubaccountContext = ({ localDydxWallet }: { localDydxWallet?: LocalWall
   ]);
 
   useEffect(() => {
-    if (referredBy && latestReferrer) {
+    if (referredBy?.affiliateAddress && latestReferrer) {
```

Comment:
[q] should we update the dependency array to only check for `referredBy?.affiliateAddress` as well?